pkg/redisstreams/redis_streams.go

```diff
@@ -269,7 +274,7 @@ func (rsSource *redisStreamsSource) xStreamToMessages(xstream redis.XStream) ([]
 }
 
 func produceMsg(inMsg redis.XMessage) (*sourcesdk.Message, error) {
-	var readOffset = sourcesdk.NewOffset([]byte(inMsg.ID), "0")
+	var readOffset = sourcesdk.NewOffset([]byte(inMsg.ID), 0)
 
```

Comment:
Since you are using DefaultPartitions() inside Partitions() method, use the default partition while creating the offset.
```suggestion
	var readOffset = sourcesdk.NewOffsetWithDefaultPartitionId([]byte(inMsg.ID))

```